internal/build/info_test.go

```diff
@@ -1,9 +1,58 @@
 package build
 
 import (
+	"context"
+	"os"
+	"path/filepath"
+	"strings"
 	"testing"
+
+	"github.com/databricks/cli/libs/git"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/mod/modfile"
 )
 
 func TestGetDetails(t *testing.T) {
 	GetInfo()
 }
+
+// This test is stored in the internal package to avoid build errors since this
+// test itself is run during the CLI generation process.
+// This test ensures that the OpenAPI SHA the CLI is being generated from matches
+// the OpenAPI SHA of the Go SDK version used in the CLI. We should always upgrade
+// the Go SDK version before generating the CLI because downstream generated assets
+// like the bundle schema depend on the Go SDK itself.
+func TestConsistentDatabricksSdkVersion(t *testing.T) {
+	// Read the go.mod file
+	b, err := os.ReadFile("../../go.mod")
+	require.NoError(t, err)
+
+	// Parse the go.mod file to get the databricks-sdk version
+	modFile, err := modfile.Parse("../../go.mod", b, nil)
+	require.NoError(t, err)
+
+	modulePath := "github.com/databricks/databricks-sdk-go"
+	var version string
+	for _, r := range modFile.Require {
+		if r.Mod.Path == modulePath {
+			version = r.Mod.Version
```

Comment:
This works if the version is a Git tag.

What happens if you use the SDK from a specific commit? (we occasionally have to do this)

Reply:
Fixed.